File: backend/EcommerceAdmin.Domain/Entities/Product.cs
namespace EcommerceAdmin.Domain.Entities;

public class Product
{
    public int Id { get; set; }
    public int CategoryId { get; set; }

    public required string SKU { get; set; }

    public required string Name {get; set; }

    public decimal Price { get; set; }

    // Navigation property
    public Category Category { get; set; } = null!;
}

File: frontend/e-commerce/src/shared/models/product.ts
import { Category } from "./category";

export interface Product {
  id: number;
  categoryId: number;
  sku: string;
  name: string;
  price: number;
  categoryName?: Category;
}

export interface CreateProductDto {
  sku: string;
  name: string;
  price: number;
  categoryId: number;
}

export class ProductParams {
    pageNumber = 1;
    pageSize = 10;
}